lib/plugins/cloud-watch.js

```diff
@@ -16,21 +21,54 @@ function register(skill, cloudwatch, eventMetric) {
     Namespace: 'StateMachine',
   };
 
-  const eventMetricConfig = _.merge(defaulEventMetric, eventMetric);
+  if (config.MetricName) {
+    _.set(defaulEventMetric, 'MetricData[0].MetricName', config.MetricName);
+  }
+
+  if (config.Namespace) {
+    _.set(defaulEventMetric, 'Namespace', config.Namespace);
+  }
+
+  const onErrorMetric = _.cloneDeep(defaulEventMetric);
+  _.set(onErrorMetric, 'MetricData[0].Value', 1);
 
-  const putMetricDataAsync = Promise.promisify(cloudwatch.putMetricData);
-  skill.onStateMachineError(alexaEvent => Promise.try(() => putMetricDataAsync(eventMetricConfig)
-  .then(() => {
+  const OnSuccessMetric = _.cloneDeep(defaulEventMetric);
+  _.set(OnSuccessMetric, 'MetricData[0].Value', 0);
+
+  const putMetricDataAsync = Promise.promisify(cloudwatch.putMetricData.bind(cloudwatch));
+
+  skill.onBeforeReplySent(alexaEvent => Promise.try(() => {
+    _.set(OnSuccessMetric, 'MetricData[0].Timestamp', new Date());
+    return putMetricDataAsync(OnSuccessMetric).then((data, error) => {
+      debug(`onBeforeReplySent Metric Data Error: ${JSON.stringify(error)}`);
+      debug(`onBeforeReplySent Metric Data Success: ${JSON.stringify(data)}`);
+      return null;
+    });
+  }));
+
+  skill.onStateMachineError(alexaEvent => Promise.try(() => {
     _.set(alexaEvent, 'cloudwatch.errorReported', true);
-    return null;
-  })));
+    _.set(onErrorMetric, 'MetricData[0].Timestamp', new Date());
 
+    return putMetricDataAsync(onErrorMetric).then((data, error) => {
+      debug(`onStateMachineError Metric Data Error: ${JSON.stringify(error)}`);
+      debug(`onStateMachineError Metric Data Success: ${JSON.stringify(data)}`);
+
+      return null;
+    });
+  }));
 
   skill.onError(alexaEvent => Promise.try(() => {
     const errorReported = _.get(alexaEvent, 'cloudwatch.errorReported');
 
     if (!errorReported) {
-      return putMetricDataAsync(eventMetricConfig).then(() => null);
+      _.set(onErrorMetric, 'MetricData[0].Timestamp', new Date());
+      return putMetricDataAsync(onErrorMetric).then((data, error) => {
+        debug(`onError Metric Data Error: ${JSON.stringify(error)}`);
```

Comment:
errors in promises don't work like that, check http://bluebirdjs.com/docs/api/then.html

`then` supports a separate error handler function, not a second `error` parameter, also, in general is better to attach a `catch` handler instead of having the second function in `then`